lib/utils.js

```diff
@@ -0,0 +1,56 @@
+const { URL } = require('url')
+
+/** this forces a minimum asterisk for security purposes */
+const asterisk = (length = 0) => {
+  if (length < 8) {
+    return '*'.repeat(8)
+  }
+  return '*'.repeat(length)
+}
+
+const redactUrlPassword = (url, replacement = asterisk()) => {
+  try {
+    const parsed = new URL(url)
+    if (parsed.password) {
+      parsed.password = replacement
+      return parsed.toString()
```

Comment:
In #6 I experimented with a solution that would only replace the password and not do any normalization via `toString()`. I think I prefer that for the client version since it is user facing and will show the url exactly as it was entered. Can we try that approach here to?

Reply:
Hah! Yeah I did think about this too, the part that makes me nervous is what if:

```
https://user:mySeceretPassword@example.com/mySeceretPassword
```

Turns into:

```
https://user:**********@example.com/**********
```

😟 

Or if the username / password are the same

```
https://**********:**********@example.com
```

There's just added complexity in ensuring that we don't expose extra information when doing an in-place password replacement. I'll look into this though and test that it doesn't have this behavior.

Reply:
You're function worked as expected! I'll bring it in!

Reply:
done